### ISSUE TITLE
This PC icons keep disappearing after reboot

### ISSUE THREAD

**Issue by imthenachoman** [madbomb122/Win10Script#6]
**Describe the bug**   
On a fresh Windows 10 install I selected "Show/Add" for all of the folders under "This PC". After I click run, before it reboots, I can see the icons when I open File Explorer. But after reboot the icons disappear again.

**To Reproduce**   
1. Select "Show/Add" for all of the folders under "This PC"
2. Run 

**Expected behavior**   
Icons should show when in File Explorer

**Screenshots**   
 ![https://i.stack.imgur.com/XAOW2.png](https://i.stack.imgur.com/XAOW2.png)

**Desktop (please complete the following information):**   
 - Windows 10 Home
 - Version 1803
 - Build 17134.472

**Additional context**
 - Script Version: 3.6.5
 - Release Type: Stable

**Comment by madbomb122** [comment]
1. On a fresh install they should be showing by default
2. the screenshot you are showing is your "user folder" not 'This PC"

**Comment by imthenachoman** [comment]
I figured out the issue. I am headed out right now but I will reply shortly with details. I think your `-force` option is deleting other keys/values that are needed. Will explain...

**Comment by madbomb122** [comment]
I tested this on windows 10 home (version 1803) in a virtual machine
what did was "Show/Add" on the fresh install, rebooted and it still showed.

I even tested "Removing" and "Hiding" the items then a "Show/Add" and it still shows the item even after reboot.

are you sure you selected "Show/Add" and not "Remove/Hide"?

Try running the script again and select "Show/Add" and reboot computer or restart explorer

**Comment by imthenachoman** [comment]
> are you sure you selected "Show/Add" and not "Remove/Hide"?

Yes. I selected "Show/Add"

> Try running the script again and select "Show/Add" and reboot computer or restart explorer

Does not work.

But I figured out the problem. A fresh installation of Windows 10 has some other keys/properties/values in the registry and your script somehow deletes those keys. For example, the downloads key, `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{7d83ee9b-2244-4e70-b1f5-5393042af1e4}\PropertyBag` also has a `dword` called `BaseFolderId` with the value `{374DE290-123F-4565-9164-39C4925E467B}`. `Win10Script` deletes that setting.

I was having the issue on two different Windows installations:

- Windows 10 Home version 1803 build 17134.472
- Windows 10 Pro version 1809 build 17763.195

My test:

1. I created a test VM with a fresh install of Windows 10 with an ISO I got from the [media creation tool](https://www.microsoft.com/en-us/software-download/windows10)
2. The icons were there
3. I made a backup of my registry
4. I ran `Win10Script` and selected "Show/Add' for all of the folders under "This PC". I know they are already shown by default on a fresh installation but I selected them just in case. In theory it shouldn't break anything if they are already there.
5. After running `Win10Script` and rebooting the icons are no longer there
6. I compared the registry and noticed a slew of the registry keys were missing the `BaseFolderId` property

This is an export of the default keys for all of the folders. You can see which ones have the `BaseFolderId` property.

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}\PropertyBag]
	"ThisPCPolicy"="Show"
	"NoCustomize"=dword:00000001

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{f42ee2d3-909f-4907-8871-4c22fc0bf756}\PropertyBag]
	"ThisPCPolicy"="Show"
	"BaseFolderId"="{FDD39AD0-238F-46AF-ADB4-6C85480369C7}"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{f42ee2d3-909f-4907-8871-4c22fc0bf756}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{7d83ee9b-2244-4e70-b1f5-5393042af1e4}\PropertyBag]
	"ThisPCPolicy"="Show"
	"BaseFolderId"="{374DE290-123F-4565-9164-39C4925E467B}"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{7d83ee9b-2244-4e70-b1f5-5393042af1e4}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{a0c69a99-21c8-4671-8703-7934162fcf1d}\PropertyBag]
	"ThisPCPolicy"="Show"
	"BaseFolderId"="{4BD8D571-6D19-48D3-BE97-422220080E43}"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{a0c69a99-21c8-4671-8703-7934162fcf1d}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{0ddd015d-b06c-45d5-8c4c-f59713854639}\PropertyBag]
	"ThisPCPolicy"="Show"
	"BaseFolderId"="{33E28130-4E1E-4676-835A-98395C3BC3BB}"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{0ddd015d-b06c-45d5-8c4c-f59713854639}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{35286a68-3c57-41a1-bbb1-0eae73d76c95}\PropertyBag]
	"ThisPCPolicy"="Show"
	"BaseFolderId"="{18989B1D-99B5-455B-841C-AB7C74E4DDFC}"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{35286a68-3c57-41a1-bbb1-0eae73d76c95}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{31C0DD25-9439-4F12-BF41-7FF4EDA38722}\PropertyBag]
	"ThisPCPolicy"="Show"

	[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{31C0DD25-9439-4F12-BF41-7FF4EDA38722}\PropertyBag]
	"ThisPCPolicy"="Show"

**Comment by madbomb122** [comment]
the only time that "BaseFolderID" is deleted is when you select the remove option

since it deletes

```
[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\XXXX

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Explorer\MyComputer\NameSpace\XXXX

[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\XXXX

[HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Explorer\MyComputer\NameSpace\XXXX
```
XXXX = being the icon ID string thing

the Show/Unhide Re-Adds the key path above (if it's not there already) and adds/changes `"ThisPCPolicy"="Show"`

I tried it multi times and even with the 'BaseFoldeId' key missing it still shows them for me (after running my script)

**Comment by imthenachoman** [comment]
But on my fresh VM install I never did a delete. I only did the Show/Add. I
will do another test VM install later to confirm.

**Comment by madbomb122** [comment]
did you see if it showed after adding the `BaseFoldeId` keys?
if it does i'll add it in my script to make sure they are there.

**Comment by madbomb122** [comment]
btw i noticed from your screenshot that the start of the navigation list folders is `Desktop`.. mine is `ThisPC` (even on the fresh install). 

Is it like that for you on a fresh install?

I wonder if that is part of the issue

**Comment by imthenachoman** [comment]
> I tried it multi times and even with the 'BaseFoldeId' key missing it still shows them for me (after running my script)

Even after reboot?

> did you see if it showed after adding the BaseFoldeId keys?

Yes

> btw i noticed from your screenshot that the start of the navigation list folders is Desktop.. mine is ThisPC (even on the fresh install).

Right click on the left area and select "Show all folders". I don't know if that should change anything...

So I just did a test by running this PowerShell script. Right now the `BaseFolderId` key is there but after running the below code it is gone.

    $Path = '\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{35286a68-3c57-41a1-bbb1-0eae73d76c95}\PropertyBag'
    New-Item -Path "HKLM:\SOFTWARE\$Path" -Force

I can only assume that the `-Force` option is overwriting it.

**Comment by madbomb122** [comment]
> > I tried it multi times and even with the 'BaseFoldeId' key missing it still shows them for me (after running my script)
> 
> Even after reboot?

Yes


> > btw i noticed from your screenshot that the start of the navigation list folders is Desktop.. mine is ThisPC (even on the fresh install).
> 
> Right click on the left area and select "Show all folders". I don't know if that should change anything...

I turned this on didn't see the folders, i undid what my script did (added the `BaseFolderId` keys) and they showed. So it seems like the `BaseFolderId` is for showing the folders in the User's Folder


> So I just did a test by running this PowerShell script. Right now the `BaseFolderId` key is there but after running the below code it is gone.
> 
> $Path = '\Microsoft\Windows\CurrentVersion\Explorer\FolderDescriptions\{35286a68-3c57-41a1-bbb1-0eae73d76c95}\PropertyBag'
> New-Item -Path "HKLM:\SOFTWARE\$Path" -Force
> ```
> I can only assume that the `-Force` option is overwriting it.

yeah it is the `-Force` option that removes it, i am adding the `BaseFolderId` keys to the script with the `Show/Add` options

**Comment by imthenachoman** [comment]
Alternatively, maybe it would be better to check if the folder exists before creating? That way if MS adds other keys/sub-folders then there won't be any issues.

**Comment by madbomb122** [comment]
I removed the `-Force`and added the `BasefolderIDs`.. When i am done with some tests I will release it with the changes

**Comment by madbomb122** [comment]
I dont think it was something that was added, i think it has been there for a while

**Comment by madbomb122** [comment]
This issue should now be Solved in v3.6.6

**Comment by imthenachoman** [comment]
:) Thanks!
